rasa/utils/tensorflow/model_data_utils.py

```diff
@@ -63,8 +68,20 @@ def featurize_training_examples(
                     attribute, featurizers
                 )
         output.append(attribute_to_features)
-
-    return output
+    # get sparse feature sizes
+    sparse_attributes = []
+    for attr, features in output[0].items():
+        if features and features[0].is_sparse():
+            sparse_attributes.append(attr)
+    # we exclude labels at this point

```

Comment:
Comments should describe the `why` of code, not the `what` of code. `what` is understandable from the code.

Reply:
Please add comment for the `why` of the code at the place where you have moved this logic to. Currently, the code calls `_remove_label_attribute....` without any explanation of why we do that.